lib/src/main/scala/spinal/lib/Mem.scala

```diff
@@ -48,9 +48,6 @@ class MemPimped[T <: Data](mem: Mem[T]) {
     val retValid = RegInit(False)
     val retData = mem.readSync(cmd.payload, cmd.fire, clockCrossing = crossClock)
 
-    when(ret.ready) {
-      retValid := Bool(false)
```

Comment:
Is there any reason to remove this block?

Reply:
Because of the **last win** rule,  the first condition `ret.ready` of `retValid` only wins when the second condition`cmd.ready` becomes false, however,  `cmd.ready` = `ret.isFree` = **False** means `ret.valid && !ret.ready`. It means when the `cmd.ready` is false,  `ret.ready` will be **False** as well. Therefore, the first condition will always be false for retValid, it should be removed.

Reply:
The function was written in a very generic way. But when we take everything toggether, yes right, that when(ret.ready) can be removed.

Reasons is : 

If ret.ready == True, then cmd.ready is True aswell (as ret.isFree = !ret.valid || ret.ready)
So, the retValid := Bool(false) will always be erased by retValid := cmd.valid

Reply:
Then I suggest split this modification into another PR, to keep this PR concentrate on specific topic.